tests/components/husqvarna_automower/common.py

```diff
@@ -0,0 +1,40 @@
+"""Husqvarna Automwoer common helpers for tests."""
+from unittest.mock import patch
+
+from homeassistant.components.husqvarna_automower.const import (
+    DOMAIN,
+    OAUTH2_AUTHORIZE,
+    OAUTH2_TOKEN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .conftest import mower_data
+from .const import TEST_CLIENT_ID, TEST_CLIENT_SECRET
+
+
+async def setup_platform(hass: HomeAssistant, mock_config_entry, side_effect=None):
+    """Set up the Husqvarna Automower platform."""
+
+    config_entry_oauth2_flow.async_register_implementation(
+        hass,
+        DOMAIN,
+        config_entry_oauth2_flow.LocalOAuth2Implementation(
+            hass,
+            DOMAIN,
+            TEST_CLIENT_ID,
+            TEST_CLIENT_SECRET,
+            OAUTH2_AUTHORIZE,
+            OAUTH2_TOKEN,
+        ),
+    )
```

Comment:
Not sure why this is needed, maybe you can use something like I did in Withings:
https://github.com/home-assistant/core/blob/b42c47e800e6be6a370d33d602948331b151980b/tests/components/withings/conftest.py#L46-L55